trie/trie.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package trie
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethdb/leveldb"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/trie"
+)
+
+// Trie is a wrapped ethereum Trie object
+type Trie struct {
+	trie     *trie.Trie
+	txStored int
+}
+
+// TxTries stores all the instances of tries we have on disk
+type TxTries struct {
+	// TODO: the memory allocated for these is hard to get back, look for better way to have a queue
+	txTries      []*Trie
```

Comment:
why not use map[common.Hash]Trie ? this will make indexOfRoot function constant time. My idea is that according to what i sought in code, these arrays could be very huge. 
Has maps will not radically increase memory usage

Reply:
txTries is now of type  map[common.Hash]Trie, indexOfRoot was removed as there is no longer a need for it, though I kept the txRoots slice as we need to maintain ordering.